### PR TITLE
[nnapi] add log_softmax

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61378**

Differential Revision: [D29597355](https://our.internmc.facebook.com/intern/diff/D29597355)

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29597355).

File: test/test_nnapi.py [hunk 1]
```diff
@@ -227,6 +227,11 @@ def test_hardtanh(self):
         with self.assertRaisesRegex(Exception, "hardtanh with args"):
             self.check(torch.nn.Hardtanh(0.0, 5.0), inp)
 
+    def test_log_softmax(self):
+        inp = torch.randn(3, 1000)
```
Comment (axitkhurana) [inline test/test_nnapi.py:231]: probably don't need 1000? let's just do a smaller number?

File: torch/backends/_nnapi/serializer.py [hunk 2]
```diff
@@ -1573,6 +1575,30 @@ def add_conv_underscore(self, node):
...
+            jit_half_to_float
+        ) = node.inputs()
+        input_id, input_oper = self.get_tensor_operand_by_jitval_fixed_size(jit_input)
+        _, dim = self.get_constant_value(jit_dim, "IntType")
+
+        out_shape = input_oper.shape
+
+        inputs = [None] * 3
+        inputs[0] = input_id
+        inputs[1] = self.add_immediate_float_scalar(1)
```
Comment (axitkhurana) [inline torch/backends/_nnapi/serializer.py:1594]: Let's add a comment on what this 1 is? "specifying the positive scaling factor for the exponent, beta."

**Review by axitkhurana — APPROVED** [review]
Looks good otherwise! Thanks for adding the op

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29597355).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29597355).

**Pull request closed without merging**